spec/std/crypto/bcrypt_spec.cr

```diff
@@ -1,50 +1,38 @@
 require "spec"
 require "crypto/bcrypt"
 
-describe "Bcrypt" do
-  it "raises if cost is to low" do
-    expect_raises ArgumentError, /Invalid cost size/ do
-      Crypto::Bcrypt.digest("secret", 3)
+describe "Crypto::Bcrypt" do
+  vectors = [
+    { 6,  "a",                                  "m0CrhHm10qJ3lXRY.5zDGO", "3rS2KdeeWLuGmsfGlMfOxih58VYVfxe" },
+    { 6,  "abc",                                "If6bvum7DFjUnE9p2uDeDu", "0YHzrHM6tf.iqN8.yx.jNN1ILEf7h0i" },
+    { 6,  "abcdefghijklmnopqrstuvwxyz",         ".rCVZVOThsIa97pEDOxvGu", "RRgzG64bvtJ0938xuqzv18d3ZpQhstC" },
+    { 6,  "~!@#$%^&*()      ~!@#$%^&*()PNBFRD", "fPIsBO8qRqkjj273rfaOI.", "HtSV9jLDpTbZn782DC6/t7qT67P6FfO" },
+    { 8,  "~!@#$%^&*()      ~!@#$%^&*()PNBFRD", "Eq2r4G/76Wv39MzSX262hu", "zPz612MZiYHVUJe/OcOql2jo4.9UxTW" },
+    { 10, "~!@#$%^&*()      ~!@#$%^&*()PNBFRD", "LgfYWkbzEvQ4JakH7rOvHe", "0y8pHKF9OaFgwUZ2q7W2FFZmZzJYlfS" },
+  ]
```

Comment:
Ugly construct (`Array` is missing `to_slice`), but `String.new(Slice.new(UInt8[0xa3].to_unsafe, 1))` if you need them as a String,  though it would probably good to accept the Slice and maybe even Array too.